documentation/quarkus/guide-how-to-create-a-devon4quarkus-app.asciidoc

```diff
@@ -0,0 +1,223 @@
+
+You can run the packaged application with this command `java -jar target/quarkus-app/quarkus-run.jar` and check http://localhost/hello.
+
+
+Let’s now produce a native executable for our application. It improves the startup time of the application and produces a minimal disk footprint. The executable would have everything to run the application including the "JVM" (shrunk to be just enough to run the application), and the application.
+
+
+Building a native executable requires using a distribution of GraalVM and a configured $GRAALVM_HOME. You can create a native application without GraalVM and use a multi-stage Docker build to run Maven inside a Docker container that embeds GraalVM. This will be explained in this chapter <<Build native executables without GraalVM>> or this https://quarkus.io/guides/building-native-image#container-runtime[guide]
+
+
+=== Installing GraalVM
+
+Building a native application requires an configured installation of GraalVM. This will be shown in this guide https://quarkus.io/guides/building-native-image#prerequisites-for-oracle-graalvm-ceee.
+
+Execute the command `mvn package -Pnative` and a native executable will be created in the target folder. Depending on the operating system you execute the command the native Linux binary *getting-started-1.0.0-SNAPSHOT-runner* or the Windows executable *getting-started-1.0.0-SNAPSHOT-runner.exe* will be created. Start the application and you should see the fast startup time and if you inspect the process properties also the low memory usage. You can again access http://localhost/hello to check the application.
+
+
+=== Build native executables without GraalVM 
+
+You can build the Linux executables without installing GraalVM or sometimes you just need the Linux executables on Windows to run them in a container. To do that you need a working container runtime, we use Docker in this guide. You can install Docker with your devonfw-ide distribution just follow this description https://github.com/devonfw/ide/blob/master/documentation/docker.asciidoc[Docker with devonfw-ide]. 
+
+[sidebar]
+.Docker on Windows
+--
+There are two modes for Docker on Windows, if you are using it in Hyper-V mode and not WSL 2 mode you have to share the project drive in the Docker settings or you will face some errors. 
+--
+We can start building our container image for the project.
+Create an executable in a container runtime with the command `mvn package -Pnative "-Dquarkus.container-image.build=true"` normally quarkus automatically detects the container runtime, but sometimes an error occurs and then you should try to you can explicitly select the container runtime with the parameter `"-Dquarkus.native.container-runtime=docker"´. You can omit the `-Pnative` parameter to create a Dockerfile with the .jar and not the native executable.
```

Comment:
...with the command `mvn package -Pnative "-Dquarkus.container-image.build=true"`. Normally Quarkus...
In the second paramter you use a front tick (if this is the right name for it), so it is not displayed properly as a command.